packages/x-charts/src/hooks/useIsHydrated.ts

```diff
@@ -5,7 +5,9 @@ import * as React from 'react';
  *
  * Basically a implementation of Option 2 of this gist: https://gist.github.com/gaearon/e7d97cdf38a2907924ea12e4ebdf3c85#option-2-lazily-show-component-with-uselayouteffect. */
 export function useIsHydrated() {
-  const [isHydrated, setIsHydrated] = React.useState(false);
+  const [isHydrated, setIsHydrated] = React.useState(
+    typeof window !== 'undefined' || process.env.NODE_ENV === 'test',
+  );
```

Comment:
@bernardobelchior this was causing issues in Vitest. Can you check if the new logic doesn't interfere with the previous behaviour?

Reply:
What issue was it causing?

Reply:
It was triggering the error that an update wasn't wrapped in `act()`

Reply:
I think that's expected since there's a `useEffect` running. Isn't the test fixed by wrapping `render` in `act?

Reply:
`render` is automatically wrapped in `act`. Idk why this happens for this specific issue 😰